### PR TITLE
Doctor side

### PR DESCRIPTION
**Pull request by Pilkas-git**
Finished Doctor side. Will still need changes when db is fully connected. 
main app changes:
Histroy displays only current user history.
group deletion.
other small code fixes.

doctor side:
added the ability to easily change ui/design of the app.
the background changes based on the gradient which is set in resource/colors.
removed camera and all features that uses it (not hsitory as it onl uses stats).

Doctors can:
Do all the previous stuff you could do with groups
Add user to group.
can check group stats, individual user stats(if thei are in their group);

all that is missing is integration to the online database.
(basicly pulling groups,user lists etc)

### CODE REVIEW

**Review by AArlauskas — APPROVED** [review]
Too long, didn't read but I trust you 😄

File: app-android-xamarin-Doctor-side/Moodis/Moodis/Feature/Menu/MenuActivity.cs [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+            var DisplayUsers = FindViewById(Resource.Id.DisplayYourGroupMembersButton);
+
+            AddUserToGroup.Click += delegate {
+                LayoutInflater layoutInflater = LayoutInflater.From(this);
+                View view = layoutInflater.Inflate(Resource.Layout.user_input_dialog_box, null);
+
+                var groupsList = GroupActivityModel.groups.Where(group => group.IsMember(SignInViewModel.currentUser.Username))
+                .Select(group => group.Groupname).ToList();
+
+                var spinner = view.FindViewById<Spinner>(Resource.Id.spinnerGroupName);
```
Comment (tmaciulis22) [inline app-android-xamarin-Doctor-side/Moodis/Moodis/Feature/Menu/MenuActivity.cs]: you could have initialized this view outside the onclick method, now you are initializing it four times in code

File: app-android-xamarin-Doctor-side/Moodis/Moodis/Feature/Menu/UserListAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        public override void OnBindViewHolder(RecyclerView.ViewHolder holder, int position)
+        {
+            (holder as UserViewHolder).OnBind(userList[position]);
+
+            (holder as UserViewHolder).ItemView.SetOnClickListener(this);
+            (holder as UserViewHolder).ItemView.Tag = position;
+            (holder as UserViewHolder).UsernameLabel.SetTextColor(userList[position].IsSelected ? Android.Graphics.Color.Green : Android.Graphics.Color.White);
+
```
Comment (tmaciulis22) [inline app-android-xamarin-Doctor-side/Moodis/Moodis/Feature/Menu/UserListAdapter.cs]: gotcha, empty line

File: app-android-xamarin-Doctor-side/Moodis/Moodis/Feature/Register/RegisterViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+using Moodis.Network.Face;
+using System;
+using System.Threading.Tasks;
+
+namespace Moodis.Feature.Register
+{
+    public class RegisterViewModel
+    {
+        public const int RequiredNumberOfPhotos = 3;
+
```
Comment (tmaciulis22) [inline app-android-xamarin-Doctor-side/Moodis/Moodis/Feature/Register/RegisterViewModel.cs]: unnecessary empty line

File: app-android-xamarin-Doctor-side/Moodis/Moodis/Models/Group.cs [hunk 4]
```diff
@@ -42,17 +44,14 @@ public void RemoveMember(string user)
 
         public bool IsMember(string username)
         {
-            return Members.Contains(username);
+            return Members.Contains(username) ? true : false;
```
Comment (tmaciulis22) [inline app-android-xamarin-Doctor-side/Moodis/Moodis/Models/Group.cs]: wont `contains` return a bool?

File: app-android-xamarin-Doctor-side/Moodis/Moodis/Resources/layout/activity_group.xml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        android:text="@string/groupName"
+        android:textSize="50dp"
+        app:layout_constraintBottom_toTopOf="@+id/text_groupName"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.497"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.576"
+        
+    />
```
Comment (tmaciulis22) [inline app-android-xamarin-Doctor-side/Moodis/Moodis/Resources/layout/activity_group.xml]: move it to last attribute

File: app-android-xamarin-Doctor-side/Moodis/Moodis/Resources/layout/activity_menu.xml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    <include
+    layout="@layout/activity_menu_constraint"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"/>
+    
+    <include
+    layout="@layout/app_bar_main"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent" />
```
Comment (tmaciulis22) [inline app-android-xamarin-Doctor-side/Moodis/Moodis/Resources/layout/activity_menu.xml]: indentation

**Review by tmaciulis22 — CHANGES_REQUESTED** [review]
Some changes

File: app-android-xamarin-Doctor-side/Moodis/Moodis/Feature/Menu/MenuActivity.cs (hunk 1 above)
Comment (Pilkas-git) [reply app-android-xamarin-Doctor-side/Moodis/Moodis/Feature/Menu/MenuActivity.cs]: Views need to be reinitialized as they keep updating the spinner changes depending on the ammount of groups the user has. If he creates 1 or deletes 1 the spinner needs to be updated and this calls for view update.

File: app-android-xamarin-Doctor-side/Moodis/Moodis/Feature/History/HistoryActivity.cs [hunk 7]
```diff
@@ -28,8 +28,8 @@ protected override void OnCreate(Bundle savedInstanceState)
             base.OnCreate(savedInstanceState);
             this.SetSupportActionBar();
             SetContentView(Resource.Layout.activity_history);
-            EXTRA_NAME = Intent.GetStringExtra("name");
-            EXTRA_REASON = Intent.GetIntExtra("reason",0);
+            EXTRA_NAME = Intent.GetStringExtra("EXTRA_NAME");
+            EXTRA_REASON = Intent.GetIntExtra("EXTRA_REASON", 0);
```
Comment (tmaciulis22) [inline app-android-xamarin-Doctor-side/Moodis/Moodis/Feature/History/HistoryActivity.cs]: no need for quotation marks if you are using string const

**Review by tmaciulis22 — CHANGES_REQUESTED** [review]
Minor change, after it you can merge and connect to BE later on

**Review by tmaciulis22 — APPROVED** [review]

**Pull request merged**